pkg/controller/backup_controller.go

```diff
@@ -148,12 +148,36 @@ func NewBackupController(
 }
 
 func (c *backupController) resync() {
+	// recompute backup_total metric
 	backups, err := c.lister.List(labels.Everything())
 	if err != nil {
 		c.logger.Error(err, "Error computing backup_total metric")
 	} else {
 		c.metrics.SetBackupTotal(int64(len(backups)))
 	}
+
+	// recompute backup_last_successful_timestamp metric for each
+	// schedule (including the empty schedule, i.e. ad-hoc backups)
+	lastSuccessBySchedule := map[string]time.Time{}
+	for _, backup := range backups {
```

Comment:
suggest pulling this loop into a `func` of its own that can be unit tested.

Reply:
sure, will do

Reply:
done, see latest commit